internal/logs/syslog_transport.go

```diff
@@ -0,0 +1,86 @@
+package logs
+
+import (
+	"errors"
+	"fmt"
+	"log/syslog"
+	"net"
+	"os"
+	"sync"
+	"time"
+)
+
+const (
+	priority = syslog.LOG_INFO
+)
+
+// We implement this way to force timestamp with the original ones.
+type syslogTransport struct {
+	server   string
+	protocol string
+	hostname string
+	conn     net.Conn
+	lock     sync.Mutex // guards conn
+}
+
+func NewSyslogServer(server string, protocol string) (*syslogTransport, error) {
+	proto := "tcp"
+	if protocol != "" {
+		proto = protocol
+	}
+	syslogInstance := &syslogTransport{
+		server:   server,
+		protocol: proto,
+		lock:     sync.Mutex{},
+	}
+	err := syslogInstance.Init()
+	if err != nil {
+		return nil, err
+	}
+	return syslogInstance, nil
+}
+
+func (s *syslogTransport) Init() error {
+	hostname, _ := os.Hostname()
```

Comment:
what if hostname isn't set for the device and it returns `localhost`?
also, should this error be ignored?

Reply:
umm, thinking about this, maybe we can set the deviceID here.